phantazm-mob/src/main/java/com/github/phantazmnetwork/mob/target/NearestEntitySelector.java

```diff
@@ -0,0 +1,81 @@
+package com.github.phantazmnetwork.mob.target;
+
+import com.github.phantazmnetwork.mob.PhantazmMob;
+import it.unimi.dsi.fastutil.objects.Object2DoubleMap;
+import it.unimi.dsi.fastutil.objects.Object2DoubleOpenHashMap;
+import net.minestom.server.entity.Entity;
+import net.minestom.server.instance.Instance;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.function.ToDoubleFunction;
+
+public abstract class NearestEntitySelector<TReturn> implements TargetSelector<Iterable<TReturn>> {
+
+    private final double range;
+
+    private final int targetLimit;
+
+    public NearestEntitySelector(double range, int targetLimit) {
+        this.range = range;
+        this.targetLimit = targetLimit;
+    }
+
+    @Override
+    public @NotNull Optional<Iterable<TReturn>> selectTarget(@NotNull PhantazmMob mob) {
+        Instance instance = mob.entity().getInstance();
+        if (instance == null) {
+            throw new IllegalStateException("instance unset");
+        }
+
+        Iterable<Entity> entities = instance.getNearbyEntities(mob.entity().getPosition(), range);
```

Comment:
this probably holds true elsewhere, but i feel using Collection<Entity> would provide more utility: that way, you can infer reasonable initial sizes for the lists/hashmaps you end up creating